File: cap2-fsharp/PeriodoDeTempo.fs
module Cap2
open System

type Periodo = { DataInicial: DateTime; DataFinal: DateTime }

// val adicionarDias: periodo: Periodo -> dias: float -> Periodo
let adicionarDias periodo dias = {
    DataInicial = periodo.DataInicial.AddDays dias
    DataFinal = periodo.DataFinal dias
}

let verificarSeDataEstaEntreOPeriodo (periodo: Periodo) (dataParaTestar: DateTime) =
    dataParaTestar.CompareTo    periodo.DataInicial >= 0
    && dataParaTestar.CompareTo periodo.DataFinal <= 0

File: dotnet-functional/cap2-csharp/PeriodoDeTempo.cs
using System;

namespace cap2
{
    public class PeriodoDeTempo
    {
        public DateTime DataInicial { get; set; }
        public DateTime DataFinal   { get; set; }

        public bool VerificarSeDataEstaEntrePeriodo(
                DateTime dataParaTestar)
        {
            return dataParaTestar.CompareTo(DataInicial) >= 0
                && dataParaTestar.CompareTo(DataFinal)   <= 0;
        }
    }
}
